src/TodoContext.tsx

```diff
@@ -0,0 +1,317 @@
+  const existingCompleted = todos.some((todo) => {
+    return todo.completed;
+  });
+
+  const nonCompletedTodos = todos.reduce((counter, todo) => {
+    if (!todo.completed) {
+      return counter + 1;
+    }
+
+    return counter;
+  }, 0);
+
+  useEffect(() => {
+    api.getTodos(USER_ID)
+      .then((apiTodos) => {
+        setTodos(apiTodos);
+        setFilteredTodos(apiTodos);
+        setTimeToFocus(current => current + 1);
+      })
+      .catch(() => {
+        setError('Unable to load todos');
+        setTimeToFocus(current => current + 1);
+      });
+  }, []);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setError('');
+    }, 3000);
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [error]);
+
+  useEffect(() => {
+    const filtereDtodo = () => {
+      switch (filter) {
+        case 'active':
+          return todos.filter((todo) => {
+            return todo.completed === false;
+          });
+        case 'completed':
+          return todos.filter((todo) => {
+            return todo.completed === true;
+          });
+        case 'all':
+          return todos;
+        default:
+          return todos;
+      }
+    };
+
+    setFilteredTodos(filtereDtodo());
+  }, [filter, todos]);
+
+  const handleSubmit = () => {
+    if (!postTodo.trim()) {
+      setDisableInput(false);
+
+      return;
+    }
+
+    const newTodo = {
+      id: 0,
+      userId: USER_ID,
+      title: postTodo.trim(),
+      completed: false,
+    };
+
+    setTempTodo(newTodo);
+
+    api.postTodos(newTodo)
+      .then((todo) => {
+        setTempTodo(null);
+        setTodos((currentTodos) => [...currentTodos, todo]);
+        setPostTodo('');
+        setTimeToFocus(current => current + 1);
+      })
+      .catch(() => {
+        setTempTodo(null);
+        setError('Unable to add a todo');
+        setTimeToFocus(current => current + 1);
+      })
+      .finally(() => {
+        titleField.current?.focus();
+        setDisableInput(false);
+      });
+  };
+
+  const handleDelete = (id: number) => {
+    setIsLoading((currentLoading) => [...currentLoading, id]);
+
+    api.deletTodos(id)
+      .then(() => {
+        setTodos((currentTodos) => currentTodos.filter(
+          (todo) => todo.id !== id,
+        ));
+        setTimeToFocus(current => current + 1);
+      })
+      .catch(() => {
+        setError('Unable to delete a todo');
+        if (titleField.current) {
+          titleField.current.focus();
+        }
+
+        setTimeToFocus(current => current + 1);
+      })
+      .finally(() => {
+        setIsLoading((currentLoading) => {
+          return currentLoading.filter((loadingId) => loadingId !== id);
+        });
+        setHandleEditing(0);
+      });
+  };
+
+  const makeTodoChange = (id: number, value: string) => {
+    const changedTodo: Pick<Todo, 'title' | 'userId'>
+      = { title: value, userId: USER_ID };
+
+    setIsLoading((currentLoading) => [...currentLoading, id]);
```

Comment:
` setIsLoading` sounds like it has bollean value , maybe rename to ` setLoadingTodos`